app/src/org/commcare/views/widgets/DecimalWidget.java

```diff
@@ -37,18 +37,15 @@
  */
 public class DecimalWidget extends StringWidget {
 
-    public DecimalWidget(Context context, FormEntryPrompt prompt, boolean secret) {
-        super(context, prompt, secret);
+    public DecimalWidget(Context context, FormEntryPrompt prompt, boolean secret, boolean compact) {
+        super(context, prompt, secret, compact);
 
         // formatting
         mAnswer.setTextSize(TypedValue.COMPLEX_UNIT_DIP, mAnswerFontSize);
         mAnswer.setImeOptions(EditorInfo.IME_FLAG_NO_EXTRACT_UI | EditorInfo.IME_ACTION_NEXT);
 
         // needed to make long readonly text scroll
         mAnswer.setHorizontallyScrolling(false);
-        if (!secret) {
```

Comment:
Why did this get deleted (both here and in `IntegerWidget`)?

Reply:
This is already there in `StringWidget` [here](https://github.com/dimagi/commcare-android/blob/compactAppearance/app/src/org/commcare/views/widgets/StringWidget.java#L67). So no point including them twice.

Reply:
👍